gcp/pubsub/src/main/scala/com/commercetools/queue/gcp/pubsub/PubSubAdministration.scala

```diff
@@ -72,7 +73,7 @@ private class PubSubAdministration[F[_]](
             Subscription
               .newBuilder()
               .setTopic(topicName.toString())
-              .setName(SubscriptionName.of(project, s"fs2-queue-$name").toString())
+              .setName(SubscriptionName.of(project, s"${configs.subscriptionNamePrefix}-$name").toString())
```

Comment:
Minor: should we omit the `-`, so that one can also omit the prefix and only go with the name?

```suggestion
              .setName(SubscriptionName.of(project, s"${configs.subscriptionNamePrefix}$name").toString())
```

Reply:
I don't like that if I want to add a `-` in the name I should set the `subscriptionNamePrefix` to something like `prefx-`. Maybe we can be smarter and trying another approach? WDYT? Maybe we can let the `PubSubConfig` be an `Option` so that we can use only the `name` when the config provided is `None` ?

Reply:
We could have an approach similar to what http4s has for its backends. The configuration has a default value `PubSubConfig.default` as you did, but the configuration might have optional values. For instance, the prefix might be an option, and if it is not here (or empty), then we just use the name. Otherwise we prefix with whatever was provided. It is also up to the caller to ensure the prefix respects the [naming convention](https://cloud.google.com/pubsub/docs/pubsub-basics#resource_names).

Something like this (assuming `subscriptionNamePrefix` is `Option[String]`):

```suggestion
              .setName(SubscriptionName.of(project, s"${configs.subscriptionNamePrefix.filter(_.nonEmpty).fold("")(pfx => s"$pfx-"}$name").toString())
```

Reply:
After talking with @AL333Z, I think his solution is the easiest and most flexible one. We should document the fact that the prefix needs to contain whatever separator the user uses.

Reply:
Refactored here: e176e938416fc66d2026dc7bfc4ec009b3b13d87 . TBH I don't love the idea I would have preferred to have a dedicated `separator` property or something similar but I see your point